File: 0118-Pascal's Triangle/0118-Pascal's Triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> result;
        for (int i = 0; i < numRows; ++i)
        {
            vector<int> path;
            path.push_back(1);
            if (i != 0)
            {
                vector<int>& prev = result[i - 1];
                for (int j = 0; j < i - 1; ++j)
                {
                    path.push_back(prev[j] + prev[j + 1]);
                }
                path.push_back(1);
            }
            
            result.push_back(path);
        }
        
        return result;
    }
};

File: 0101-0200/0118-Pascal's Triangle/0118-Pascal's Triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        result = []
        for i in range(numRows):
            row = [1] * (1 + i)
            for j in range(1, i):
                row[j] = result[-1][j - 1] + result[-1][j]
            result.append(row)
        return result
